oraclepaas/resource_mysql_service_instance.go

```diff
@@ -0,0 +1,662 @@
+package oraclepaas
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	opcClient "github.com/hashicorp/go-oracle-terraform/client"
+	"github.com/hashicorp/go-oracle-terraform/mysql"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceOraclePAASMySQLServiceInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOraclePAASMySQLServiceInstanceCreate,
+		Read:   resourceOraclePAASMySQLServiceInstanceRead,
+		Update: resourceOraclePAASMySQLServiceInstanceUpdate,
+		Delete: resourceOraclePAASMySQLServiceInstanceDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(120 * time.Minute),
+			Delete: schema.DefaultTimeout(120 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"service_description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"ssh_public_key": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"availability_domain": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"backup_destination": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  string(mysql.ServiceInstanceBackupDestinationNone),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(mysql.ServiceInstanceBackupDestinationBoth),
+					string(mysql.ServiceInstanceBackupDestinationOSS),
+					string(mysql.ServiceInstanceBackupDestinationNone),
+				}, true),
+			},
+
+			"metering_frequency": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  string(mysql.ServiceInstanceMeteringFrequencyHourly),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(mysql.ServiceInstanceMeteringFrequencyHourly),
+					string(mysql.ServiceInstanceMeteringFrequencyMonthly),
+				}, true),
+			},
+
+			//			"enable_notification": {
+			//				Type:     schema.TypeBool,
+			//				Optional: true,
+			//				ForceNew: true,
+			//			},
+
+			//			"notification_email": {
+			//				Type:     schema.TypeString,
+			//				Optional: true,
+			//				ForceNew: true,
+			//			},
+
+			// Use for OCI configuration (not OCI-Classic)
+			"ip_network": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"vm_user": {
+				// default to opc
+				Type:      schema.TypeString,
+				Optional:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+
+			"cloud_storage_configuration": {
```

Comment:
This is for backups right? The other service instances call this block `backups` so let's maintain that consistency here